generator/src/main/java/com/scottlogic/deg/generator/Guice/GenerationEngineProvider.java

```diff
@@ -0,0 +1,31 @@
+package com.scottlogic.deg.generator.Guice;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.google.inject.name.Named;
+import com.scottlogic.deg.generator.CommandLine.GenerateCommandLine;
+import com.scottlogic.deg.generator.GenerationEngine;
+
+public class GenerationEngineProvider implements Provider<GenerationEngine> {
+
+    private GenerateCommandLine commandLine;
+    private GenerationEngine generationEngine;
+    private GenerationEngine violationEngine;
+
+    @Inject
+    public GenerationEngineProvider(GenerateCommandLine commandLine,
+                                    @Named("valid") GenerationEngine generationEngine,
```

Comment:
Bringing up the old debate again.

I Think we should avoid using @ Named for things that don't need to be named.

Named is primarily meant for when an injected class can have different instances in the system.

In this case, we are only using 1 possible instance of ValidGenerationEngine. and 1 possible instance of InvalidGenerationEngine.

So they can be injected in without needing @ Named.

Reply:
I disagree as this ties the provider into requiring either 
1. tests to new-up concrete dependencies and all their descendant dependencies _OR_
1. tests to use another library to overcome the above

Using the `@Named` approach overcomes this without a huge conceptual overhead.

It also allows for IoC to bind to a different 'violating' or 'normal' engine without having to change the provider at the same time.

Reply:
Leaving as is for now, can be changed in the future if needed

Reply:
👍 leave as is for now.

An issue has been raised for discussing the issue

#495